packages/@aws-cdk/aws-codepipeline-actions/lib/lambda/invoke-action.ts

```diff
@@ -43,22 +43,6 @@ export interface LambdaInvokeActionProps extends codepipeline.CommonActionProps
    */
   readonly userParameters?: any;
 
```

Comment:
Good catch on the docs, I'll update those.

The thing is, this field is more general than just a key-value pair - if it was, we could have used the `{ [key: string]: any }` type that we use, for example, for parameter overrides in the CloudFormation CodePipeline Actions. You can put literally anything in here as a string, and it's up to the Lambda to interpret it correctly.

I think a union-like class is way too heavy for this. If you don't like the solution I've proposed, I would rather just turn this property into a String, and make the customer do the `JSON.stringify()` call.

Reply:
@eladb let me know what you prefer based on my above comment.

Reply:
@eladb ping

Reply:
> You can put literally anything in here as a string, and it's up to the Lambda to interpret it correctly.

If I put something that cannot be parsed as JSON, what would happen? If this is going to fail, then what's the reason to enable?

I think `[string]: any` is actually an anti-pattern, which we should probably forbid through awslint rules (@shivlaks implemented an "any" rule, and I wonder if perhaps we need to extend it to "any[]" and "{ [string]: any }".

Reply:
Haven't quite implemented the awslint rule that forbids `any` usage, but it might be one of the next rules I get to.

The awslint rule you might be thinking of might be the one to forbid Typescript unions (#2509)

Reply:
> If I put something that cannot be parsed as JSON, what would happen? If this is going to fail, then what's the reason to enable?

Nothing will happen. If you pass a string here, it will be passed unchanged (see the test that does  that with an XML document).

Reply:
I don't think a `string` type is the right approach since I believe it's not optimizing for the common use case, which is what we should optimize for. I think the common use case would be to supply a set of key-values.

How about this: let's make `userParameters` a `{ [key: string]: any }` (yes, "any") for now. If in the future someone will come to us and say they want to pass in an arbitrary string, we can always add `userParametersString` as an additional optional property and runtime-check that they are mutually exclusive. I am willing to put money that that will never happen.

Reply:
Not a lot of money though ;)

Reply:
Changed to `{ [key: string]: any }` in the newest revision.